tests/test_tokenizer/test_tokenizer.py

```diff
@@ -50,6 +50,27 @@ def test_CharTokenizer():
     assert ret == ans
 
 
+def test_Tokenizer():
+    items = [{
+        "stem":
+        "The stationery store has $600$ exercise books, and after selling some,\
+        there are still $4$ packs left, $25$ each, how many are sold?",
+    }]
+    ans = [
+        'The', 'stationery', 'store', 'has', '$', '600', '$', 'exercise',
+        'books', 'and', 'after', 'selling', 'some', 'there', 'are', 'still',
+        '$', '4', '$', 'packs', 'left', '$', '25', '$', 'each', 'how', 'many',
+        'are', 'sold'
+    ]
+    for tok in ['nltk', 'spacy']:
+        tokenizer = get_tokenizer("char",
```

Comment:
Use `pure_text` instead of `char`

`CharTokenizer does` not have the parameter `text_params` so that it will not use nltk/spacy as you want. Go through `EduNLT/Tokenizer/tokenizer.py` carefully to understand how those tokenizers work

Reply:
See here: https://github.com/bigdata-ustc/EduNLP/actions/runs/8221030006/job/22496017488?pr=158#step:5:356

The pytest result means that line74-116 in this file is not covered by your test.